templates/vault/0/rancher-compose.yml

```diff
@@ -0,0 +1,67 @@
+version: '2'
+catalog:
+  name: Vault
+  version: 1.0-test-rancher1
+  description: |
+    Installs a Vault container, internal (Rancher) load-balancer, and allows connection to a Consul cluster
+  uuid: vault-rac-0
+  questions:
+    - variable: VAULT_LOCAL_CONFIG
+      label: Provide a default config?
+      description: |
+        This is the embedded JSON format config file that vault will use at startup
+      type: multiline
+      default: |
+        {"backend":{"consul":{"address":"http://<listenAddress>:<listenPort>", "advertise_addr":"http://<advertiseAddress>","path":"<Path>"}}, "listener":{"tcp":{"address":"0.0.0.0:18200","tls_disable":1}}}
+    - variable: useConsul
+      label: Would you like to back Vault with Consul?
+      description: |
+        Selecting yes or no allows you to back your vault repoository with a pre-installed
+        Consul cluster/server
+      default: false
+      type: boolean
+    - variable: consulService
+      label: Specify the Consul service to connect to
+      description: |
+        The pre-installed Consul server to back to
+      default: "select"
+      type: service
+    - variable: volumeDriver
+      label: Driver for volumes
+      description: |
+        How/Where to store your vault config and logs
+      default: local
+      type: enum
+      options:
+        - local
+        - rancher-nfs
+        - rancher-efs
+        - rancher-ebs

```

Comment:
What do you think to add `VAULT_LISTEN_PORT` and `VAULT_CLUSTER_PORT` to avoid collisions and be able to choose distinct ports numbers.
```
+    - variable: "VAULT_LISTEN_PORT"
+      label: "vault port"
+      description: |
+        Set port where to listen vault server.
+      required: true
+      type: "int"
+      default: 8200
+    - variable: "VAULT_CLUSTER_PORT"
+      label: "cluster port"
+      description: |
+        Set port where to listen vault cluster service.
+      required: true
+      type: "int"
+      default: 8201
```

Reply:
Again... limited on how to address this issue due to the port being hard-coded in the container itself.  But, I'm pretty sure my work-around should help address it.

Reply:
So, to wrap-up... the ports specified in the Catalog.Questions section are set on the LoadBalancer.  The "official" Vault container hard-codes these ports internally, so setting custom ports on the LB and directing traffic to the Vault (UNEXPOSED) hard-coded default ports should work; at least it does in my environment.  I can spin up 2 stacks with these changes giving port 9200 in one stack, and 8200 in the other (and different folder names in Consul) and both servers worked without conflict.